File: SalesCrawler/Scrapers/TestBikeBolha.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using SalesCrawler.Models;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Support.UI;

namespace SalesCrawler.Scrapers
{
    public class TestBikeBolha : Helpers.ScraperBase, Helpers.IScraper
    {
        // Same as BikeMag!
        public Scraper Datasheet { get; } = new Scraper()
        {
            Name = "TestBike"
        };

        public void StartSearch(ScraperSetting scraperSettings)
        {
            if (scraperSettings.IsSearchPatternURL)
            {
                driver.Navigate().GoToUrl(scraperSettings.SearchPattern);
            }
            else
            {
                driver.Navigate().GoToUrl($"https://bolha.testbike.hu/aprok?q={System.Web.HttpUtility.UrlEncode(scraperSettings.SearchPattern)}");
            }
        }

        public IReadOnlyCollection<IWebElement> GetItemsOnPage()
        {
            Waitfor(By.XPath("//h5"));
            return driver.FindElements(By.XPath("//figure"));
        }

        public void GetItem(IWebElement item, MatchData md)
        {
            var link = item.FindElement(By.XPath(".//a"));

            md.Seller = null;
            md.Title = link.GetAttribute("title");
            md.Url = link.GetAttribute("href");
            md.ImageBinary = GetImage(item.FindElement(By.XPath(".//img")).GetAttribute("src"));
            md.Description = null;
            md.ActualPrice = StripToInt(item.FindElement(By.XPath(".//span[@class='d_block']")).Text);
            md.Currency = GetCurrency(item.FindElement(By.XPath(".//span[@class='d_block']")).Text);
            md.IsAuction = false;
            md.Location = item.FindElement(By.XPath(".//span[@class='d_block fw_bold color_light']")).Text;

            md.Expire = NEVEREXPIRE;
        }

        public By NextPageElement { get; } = By.XPath("i[@class='fa fa-angle-right d_inline_m']");


        public void UpdateMatchDetails(MatchData md)
        {
            Waitfor(By.XPath("//div[@itemprop='description']"));
            md.Description = driver.FindElement(By.XPath("//div[@itemprop='description']")).Text;
            var sell = driver.FindElements(By.XPath("//a[@title='profiloldal']"));
            if (sell.Count == 0)
            {
                md.Status = MatchDataStatus.Sold;
                return;
            }
            md.Seller = sell[0].Text;
        }
    }
}

File: salesCrawlerScrapy/spiders/testbikebolha.py
import scrapy

from salesCrawlerScrapy.helpers import Helpers
from salesCrawlerScrapy.items import ProductItem
import logging

class Testbikebolha(scrapy.Spider):
    name = 'testbikebolha'
    url_for_searchterm = 'https://bolha.testbike.hu/aprok?q={searchterm}&p=1&s=cheap'
                          
    def __init__(self, searchterm=None, fullink=None, spiderbotid = -1, maxpages=15, minprice=0, maxprice=Helpers.MAXPRICE, *args, **kwargs):
        super(Testbikebolha, self).__init__(*args, **kwargs)
        if searchterm:
            self.start_urls = [Testbikebolha.url_for_searchterm.format(searchterm=searchterm, minprice=minprice, maxprice=maxprice)]
            
        if fullink:
            self.start_urls = [f'{fullink}']
        logging.debug(f"Start url is: {self.start_urls}")
        
        if type(spiderbotid) == str:
            self.spiderbotid = int(spiderbotid)
        else: 
            self.spiderbotid = spiderbotid
        
        self.maxpages=maxpages
        self.scrapedpages=0
    
    def parse(self, response):
        logging.debug(f"Parse started")
        itemcount = 0
        for item in response.xpath("//figure"):
            itemcount += 1
            logging.debug(f"Parsing item {itemcount}")
            
            yield ProductItem(
                title = Helpers.getString(item.xpath("./div[@class='relative']/div/a/@title").get()),
                url = response.urljoin(item.xpath("./div[@class='relative']/div/a/@href").get()),
                image_urls = Helpers.imageUrl(None, item.xpath("./div/div/a/img/@src").get()),
                extraid = item.xpath("./div[@class='relative']/div/a/@href").get(),
                seller = None,
                price = Helpers.getNumber(item.xpath(".//span[@class='d_block']/text()").get()),
                currency = Helpers.getCurrency(item.xpath(".//span[@class='d_block']/text()").get()),
                location = Helpers.getString(item.xpath(".//span[@class='d_block fw_bold color_light']/text()").get()),

                spiderbotid = self.spiderbotid
            )

        next_page = response.xpath("//a[i/@class='fa fa-angle-right d_inline_m']/@href").get()
        if next_page and self.scrapedpages<self.maxpages:
                self.scrapedpages += 1
                logging.debug(f"Next page (#{str(self.scrapedpages)} of {self.maxpages})")
                yield response.follow(next_page, self.parse)